File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency/Persistence/src/main/java/api/Table.java
package api;

import domain.Entity;

import java.util.List;
import java.util.Optional;

public interface Table<ID, T extends Entity<ID>> {
    void add(T elem);
    void delete(ID id);
    void update(T elem, T newElem);
    Optional<T> findById(ID id);
    List<T> getAll();
}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Repository/Database/Table.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.src.Domain;

namespace travelAgency2.Repository.Database
{
    internal interface Table<ID, T> where T : Entity<ID>
    {
        void Add(T elem);
        void Delete(ID id);
        void Update(T elem, T newElem);
        T FindById(ID id);
        List<T> GetAll();
    }
}
